src/event-bridge/types.ts

```diff
@@ -13,6 +13,7 @@ export interface Event<
   time: string;
   region: string;
   resources: string[];
+  "trace-header": string;
```

Comment:
Do you know if trace-header can be used in the pattern match and input transform?

Reply:
It does not appear so. From the [docs](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-related-service-xray.html):

> ⚠️ Important
The trace header is not available on the event that's delivered to the invocation target.

I also tested it manually, using this event pattern:
```
{
  "trace-header": [{
    "exists": true
  }]
}
```

This caused all events to be ignored, which leads me to believe that `trace-header` is dropped before _any_ processing, not just upon delivery, per the docs.

Reply:
In this case, I think we should just put the trace-header on the event that is used in the `putEvents()` integration method (and not the `Event` object used by Pattern and Input Transform).

https://github.com/functionless/functionless/blob/main/src/event-bridge/event-bus.ts#L562

```ts
export type PutEventInput<Evnt extends Event> = Partial<Evnt> &
  Pick<Evnt, "detail" | "source" | "detail-type"> & {
    "trace-header": string;
  };
```